assign2/src/presesnter/NewPostPresenter.ts

```diff
@@ -0,0 +1,47 @@
+import {
+    doAddTagToSelectedTags,
+    doClearNewPostData,
+    doNewPost,
+    doSetCurrentTag,
+    doSetNewField
+} from "../model/question/actions";
+import {Dispatch} from "redux";
+import User from "../model/objects/User";
+import Tag from "../model/objects/Tag";
+import {NewPostField} from "../model/question/types";
+import {doCreateNewTag, doEditNewTagName} from "../model/tag/actions";
+
+export const newPostPresenter = (dispatch: Dispatch) =>
+    ({
+        handleInputChange: (field: NewPostField, value: string) => {
+            dispatch(doSetNewField(field, value));
+        },
+
+        handleResetState: () => {
+            dispatch(doClearNewPostData());
+        },
+
+        handleCreatePost: (author?: User) => {
+            //FIXME Where to handle this?
+            if (author) {
```

Comment:
It is fine to do it in the presenter. For example you can show an error message because the user is not logged in.